packages/cli/src/telemetry.ts

```diff
@@ -0,0 +1,146 @@
+/// <reference lib="dom" />
+import fs from 'fs-extra';
+import uuid from 'uuid/v4';
+import path from 'path';
+import crypto from 'crypto';
+import envPaths from 'env-paths';
+import mapValues from 'lodash.mapvalues';
+import pickBy from 'lodash.pickby';
+import inquirer from 'inquirer';
+import firebase from 'firebase/app';
+import 'firebase/auth';
+import 'firebase/firestore';
+import { DISABLE_INTERACTIVITY } from './prompts/prompt';
+
+import ProjectFile from './models/files/ProjectFile';
+
+const FIREBASE_CONFIG = {
+  apiKey: 'AIzaSyAZoGB2p26UejODezZPmczgwehI6xlSKPs',
+  authDomain: 'cli-telemetry.firebaseapp.com',
+  databaseURL: 'https://cli-telemetry.firebaseio.com',
+  projectId: 'cli-telemetry',
+  storageBucket: '',
+  messagingSenderId: '449985678611',
+  appId: '1:449985678611:web:88b411adc68e6521e19ee6',
+};
+
+type Field = string | number | boolean;
+
+interface GlobalTelemetryOptions {
+  optIn: boolean;
+  uuid: string;
+  salt: string;
+}
+
+export default {
+  async report(commandName: string, options: any, interactive: boolean): Promise<void> {
+    const telemetry = await checkOptIn(interactive);
+    if (telemetry === undefined || !telemetry.optIn) return;
+
+    // extract network name if present
+    let { network } = pickBy(options, (_, key) => key === 'network');
+    if (network) delete options.network;
+    if (network && network.match(/dev-/)) network = 'development';
+
+    // encrypt data before sending it
+    const concealedData = concealData(commandName, options, telemetry.salt);
+    const commandData = network ? { ...concealedData, network } : concealedData;
+
+    await this.sendToFirebase(telemetry.uuid, commandData);
+  },
+
+  async sendToFirebase(uuid: string, commandData: any): Promise<void> {
```

Comment:
Would really appreciate having type here for `commandData` and `concealData` function return type.